doc/hooks/versionCheckHook.section.md

```diff
@@ -22,14 +24,64 @@ stdenv.mkDerivation (finalAttrs: {
 })
 ```
 
-Note that for [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function), `doInstallCheck` is enabled by default.
-
 It does so in a clean environment (using `env --ignore-environment`), and it checks for the `${version}` string in both the `stdout` and the `stderr` of the command. It will report to you in the build log the output it received and it will fail the build if it failed to find `${version}`.
 
-The variables that this phase control are:
+Note that `doInstallCheck` is enabled by default for [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function).
+
+## Attributes controlling versionCheckHook {#versioncheckhook-attributes-controlling}
+
+The following attributes control `versionCheckHook`:
+
+- `dontVersionCheck`: When set as `true`, the hook is not added to the [`preInstallCheckHooks`](#ssec-installCheck-phase).
+
+  Useful when one wants to load the function defined by the hook, so that it can be used for a different purpose.
+
+  Defaults to false.
+
+- `versionCheckProgram`: The full path to the program to be executed.
+
+   Defaults to `${placeholder "out"}/bin/${pname}`, roughly speaking.
+
+   The usage of `placeholder "out"` instead of `$out` here is unavoidable, since the hook does not expand environment variables.
+
+- `versionCheckProgramArg`: The argument to be passed to `versionCheckProgram`.
+
+  When not defined, the hook tries `--help` first and then `--version`.
+
+  Examples: `version`, `-V`, `-v`.
+
+- `preVersionCheck`: Hook that runs before the check is done. Defaults to an empty string.
+
+- `postVersionCheck`: Hook that runs after the check is done. Defaults to an empty string.
+
+Note: see also [`testers.testVersion`](#tester-testVersion).
+
+## Comparison between `versionCheckHook` and `testers.testVersion` {#versioncheckhook-comparison-testversion}
+
+The most notable difference between `versionCheckHook` and `testers.testVersion` is that `versionCheckHook` runs as a phase during the build time of the package, while `passthru.tests.versions` executes a whole derivation that depends on the package.
+
+Below we list some advantages of `testers.testVersion` over `versionCheckHook`:
+
+- `testers.testVersion` is a derivation, having access to the whole features provided by Nix and Nixpkgs
+  - E.g. extra tools, customized programs etc.
+- `testers.testVersion` executed in an identical environment of a consumer would, independent of the environment it was built.
+- `testers.testVersion` can be executed without rebuilding the package, saving time especially when the rebuild takes too much time.
+- `testers.testVersion` do not add overhead to each build, since it runs after the build, not during it.
```

Comment:
Calling this an overhead is an overstatement IMO.

Reply:
Objectively the overhead of testVersion is zero.

I will add an extra bullet.